x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/DateTimeArithmeticOperation.java

```diff
@@ -133,7 +112,7 @@ public final Object fold() {
             } catch (ArithmeticException e) {
                 // Folding will be triggered before the plan is sent to the compute service, so we have to handle arithmetic exceptions
                 // manually and provide a user-friendly error message.
-                throw IllegalTemporalValueException.fromArithmeticException(source(), e);
+                throw DateMathException.fromArithmeticException(source(), e);
```

Comment:
ExceptionUtils.math(ArithmeticException).

Reply:
I am not really sure how an `ExceptionUtils` class should look like to properly use it in this PR. As [mentioned above](https://github.com/elastic/elasticsearch/pull/99711#discussion_r1335691043), I need a new kind of `EsqlClientException`.

Reply:
```
ExceptionUtils.math(Expression e, ArithmeticExpression ex) {
   return new ParsingException(e.source, "Expression [{}] causes arithmetic exception [{}], e, ex);
} 
```

Reply:
Alright, that should work!